### PR TITLE
Dated tasks correct impl

### PR DESCRIPTION
**Pull request by gisikw**
Better (read: working) version of #4996.

Sort dated todos; uses custom filter to ensure `orderBy` is only called for dated todos; karma specs!

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Alys, did you want to do any testing on this before I merge it in?

**Comment by Alys** [comment]
No, I'm confident that nothing could possibly go wrong. :)

I wouldn't be able to test for at least 24 hours, so if you're happy with it, I reckon you should merge it.

**Comment by Alys** [comment]
Merged! Thank you! Noted towards Tier 6.

**Comment by Alys** [comment]
Actually, sorry, Tier 6 awarded. It's definitely time.

**Comment by paglias** [comment]
@gisikw hey! we're having some problems with this PRs, specifically it prevents habits, dailies and todos from displaying:

selezione_010

**Comment by gisikw** [comment]
Can I get the top of that stack trace? It wasn't happening in pre-merge testing.

Pulling down latest develop.

**Comment by paglias** [comment]
@gisikw sorry for the truncated error, i took it in a hurry because it was on production. I can't replicate locally but on production it was definitively there and I can't put it online again :)

And I'm quite sure it referred to the `conditionalOrderBy` filter.

**Comment by gisikw** [comment]
I'm spinning up my local instance now - I'm not seeing the issue on beta, though I don't know whether the code exists there?

**Comment by paglias** [comment]
@gisikw no beta is used to test some other code and not up to date with develop, and I can't put develop on it right now (cc @Alys). 

I'm wondering if it could be caused by some caching issue or error in the javascript compilation. I'll try replicating the production environment locally but it'll take some time.

**Comment by gisikw** [comment]
:-/ Yeah, I can't seem to repro locally. I would guess that yeah, either there's some caching stuff, or perhaps theres an issue somewhere in the minification with multi-arg filters? (it's really terrible syntax, using `:` as arg separators, but _shrugs_).

I can put together a revert PR if you like, though I'm assuming you've already got that?

**Comment by paglias** [comment]
@gisikw don't worry, when even tests are not enough we can't blame anyonw :)

Before reverting I want to put it on beta, but it may happen in a few hours, I'll let you know

**Comment by gisikw** [comment]
Cool cool - just wanted to make sure it wasn't blocking deploys of other important stuff.

**Comment by paglias** [comment]
@gisikw nothing critical, but if you want I found the way to reproduce:
1. here https://github.com/HabitRPG/habitrpg/blob/develop/website/src/middleware.js#L160 change the `if` so that it always evaluates to true: `ìf(true){`
2. run `grunt build:prod` (you may have to install grunt-cli: `ǹpm install -g grunt-cli`)
3. `npm start`

the site won't look good because it misses some production things but the error is there:

selezione_011

File: website/public/js/filters/filters.js [hunk 1]
```diff
@@ -21,3 +21,11 @@ angular.module('habitrpg')
         (gp > 999) ? (gp / Math.pow(10, 3)).toFixed(1) + "k" : gp;
     }
   })
+  .filter('conditionalOrderBy', function($filter) {
```
Comment (paglias) [inline website/public/js/filters/filters.js:24]: this isn't minification ready, you should use the array notation `['$filter', function($filter){}]` see https://github.com/HabitRPG/habitrpg/blob/b503a0fd641b7dc9f031b65c8dcd7f80be7b81bd/website/public/js/services/challengeServices.js#L8 for an example

Comment (gisikw) [inline website/public/js/filters/filters.js:24]: DANGIT! There it is >.<

Lemme get that fixed up. Ugh.

**Comment by Alys** [comment]
Sorry everyone, I've just woken up. @paglias, send to beta whenever you want.

**Pull request merged**